common/doc_store/infinity_conn_base.py

```
@@ -789,20 +789,23 @@ def get_highlight(self, res: tuple[pd.DataFrame, int] | pd.DataFrame, keywords:
+            if field_name == "content_with_weight" and "content" in df.columns:
                 field_name = "content"
             else:
                 return {}
+        if "id" not in df.columns:
+            return {}
+
+        ans = {}
+        num_rows = len(df)
+        column_id = df["id"]
         for i in range(num_rows):
             id = column_id[i]
```

Ruff (0.16.2)

[error] Variable `id` is shadowing a Python builtin

(A001)